src/actions/receive-profile.js

```diff
@@ -1051,6 +1051,26 @@ export function retrieveProfileFromBrowser(
       });
       const unpackedProfile =
         await _unpackGeckoProfileFromBrowser(rawGeckoProfile);
+      const meta = unpackedProfile.meta;
+      if (meta.configuration.features.includes('power')) {
```

Comment:
`meta.configuration` was added later and [it's an optional property](https://github.com/firefox-devtools/profiler/blob/39cde2839db63467c804d5fb01cc5154cacf8e78/src/types/gecko-profile.js#L380-L381) of `meta`. Could you add a check to make sure that we have this property like `if (meta.configuration && meta.configuration.features.includes('power'))`.

Normally flow would warn us for cases like this, but we are very early in the profile retrieval here. That's why the type of `unpackedProfile` is still `MixedObject`.

Reply:
Thanks for the explanation of why flow wasn't helpful here! :)